spec/models/utils/bulk_media_packager_spec.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+require "fileutils"
+
+describe Utils::BulkMediaPackager do
+  let(:user) { create(:user, role_name: "coordinator") }
+  let(:operation) { create(:operation, creator: user) }
+  let!(:form) { create(:form, name: "foo", question_types: %w[image]) }
+  let!(:form2) { create(:form, name: "bar", question_types: %w[image]) }
+
+  let(:media_jpg) { create(:media_image, :jpg) }
+  let(:media_png) { create(:media_image, :png) }
+  let(:media_tiff) { create(:media_image, :tiff) }
+
+  let!(:responses) do
+    [
+      create(:response, form: form, answer_values: [media_jpg]),
+      create(:response, form: form, answer_values: [media_png]),
+      create(:response, form: form2, answer_values: [media_tiff])
+    ]
+  end
+
+  describe "bulk image packager" do
+    it "should return the correct size of images" do
+      ability = Ability.new(user: operation.creator, mission: operation.mission)
+      packager = described_class.new(ability: ability, search: nil, operation: operation)
+      size = packager.calculate_media_size
+      expect(size).to equal(1_819_435)
+    end
+
+    it "should return the correct size of images with search" do
+      ability = Ability.new(user: operation.creator, mission: operation.mission)
+      packager = described_class.new(ability: ability, search: "form: foo", operation: operation)
+      size = packager.calculate_media_size
+      expect(size).to equal(855_939)
+    end
+
+    it "should download, zip all the images, and cleanup" do
+      ability = Ability.new(user: operation.creator, mission: operation.mission)
+      packager = described_class.new(ability: ability, search: "form: foo", operation: operation)
+      results = packager.download_and_zip_images
+
+      expect(results.basename.to_s).to match(/bulk_images_archive/)
+      expect(File.exist?(results.to_s)).to be(true)
+      expect(Dir["#{results.dirname}/*.jpg"].any?).to be(false)
```

Comment:
At what point do these temporary images actually get removed? I assume it's happening automatically when you include them in the zip file or something? Just curious.

Reply:
that is what the clean up method in the gem is for. in the gem nothing is called in clean up, so that is why I extended it.

Reply:
Ohhh neat! Thanks for explaining